inference-engine/src/inference_engine/network_serializer.cpp

```
@@ -4,165 +4,265 @@
-        for (const auto & op : function->get_ops()) {
-            auto & rtInfo = op->get_rt_info();
-            if (rtInfo.find(ExecGraphInfoSerialization::PERF_COUNTER) == rtInfo.end()) {
-                execGraphInfoSerialization = false;
-                break;
-            }
-        }
+ConstantAtributes dump_constant_data(std::vector<uint8_t> &bin,
+                                     const ngraph::op::Constant &c) {
+  ConstantAtributes attr;
+  const uint8_t *p = reinterpret_cast<const uint8_t *>(c.get_data_ptr());
+  attr.size = ngraph::shape_size(c.get_shape()) * c.get_element_type().size();
+  attr.offset = bin.size();
+  bin.insert(end(bin), p, p + attr.size);
+  return attr;
+}
+
+void ngfunction_2_irv10(pugi::xml_document &doc, std::vector<uint8_t> &bin,
+                        const ngraph::Function &f) {
+  pugi::xml_node netXml = doc.append_child("net");
+  netXml.append_attribute("name").set_value(f.get_friendly_name().c_str());
+  netXml.append_attribute("version").set_value("10");
+  pugi::xml_node layers = netXml.append_child("layers");
+
+  auto layer_ids = create_layer_ids(f);
+  for (auto node : f.get_ordered_ops()) {
+    // <layers>
+    pugi::xml_node layer = layers.append_child("layer");
+    layer.append_attribute("id").set_value(layer_ids[node.get()]);
+    layer.append_attribute("name").set_value(node->get_friendly_name().c_str());
+    layer.append_attribute("type").set_value(node->get_type_name());
+    // TODO: calculate real opset
+    layer.append_attribute("version").set_value("opset1");
 
-        if (execGraphInfoSerialization) {
-            pugi::xml_document doc;
-            FillXmlDocWithExecutionNGraph(network, doc);
+    // <layers/data>
+    pugi::xml_node data = layer.append_child("data");
+
+    // <layers/data> general atributes
+    XmlVisitor visitor{data};
+    try {
+      if (!node->visit_attributes(visitor)) {
+        #if 0 // TODO: need to decide if we want to throw on vistiation fail
```

I think that we can throw an exception in this case, because for serialization and deserialization all nodes should implement visit_attribute method and just add visit_attributes to `ExecutionNode` class.

---

Done in this [commit](https://github.com/openvinotoolkit/openvino/pull/2579/commits/8610ed93d0e2349e5a1508a018a5efd9b697e7fa)